Makefile

```diff
@@ -54,17 +52,42 @@ ifeq ($(GOOS), linux)
 endif
 
 GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD 2>/dev/null)
-IMAGE := skopeo-dev$(if $(GIT_BRANCH),:$(GIT_BRANCH))
-# set env like gobuildtag?
-CONTAINER_CMD := ${CONTAINER_RUNTIME} run --rm -i -e TESTFLAGS="$(TESTFLAGS)" #$(CONTAINER_ENVS)
+
+# If $TESTFLAGS is set, it is passed as extra arguments to 'go test'.
+# You can increase test outputerbosity with the option '-test.vv'.
+# You can select certain tests to run, with `-test.run <regex>` for example:
+#
+#     make test-unit TESTFLAGS='-test.run ^TestBuild$'
+#
+# For integration test, we use [gocheck](https://labix.org/gocheck).
+# You can increase test outputerbosity with the option '-check.vv'.
+# You can limit test selection with `-check.f <regex>`, for example:
+#
+#     make test-integration TESTFLAGS='-check.f DockerSuite.TestBuild*'
+#
+export TESTFLAGS ?= -v -check.v -test.timeout=15m
+
+# This is assumed to be set non-empty when operating inside a CI/automation environment
+CI ?=
+
+# This env. var. is interpreted by some tests as a permission to
+# modify local configuration files and services.
+export SKOPEO_CONTAINER_TESTS ?= $(if $(CI),1,0)
+
+# This is a compromise, we either use a container for this or require
+# the local user to have a compatible python3 development environment.
+# Define it as a "resolve on use" variable to avoid calling out when possible
+export SKOPEO_CIDEV_CONTAINER_FQIN ?= $(shell hack/get_fqin.sh)
```

Comment:
It turns out this doesn’t work, the script is run even if not used (which I noticed because macOS doesn’t have `realpath(1)`, sigh).

Removing `export` seems to help; and because we only use `SKOPEO_CIDEV_CONTAINER_FQIN` to start a container, not to do anything inside the container, it seems to me that removing `export` shouldn’t cause any extra script invocations.

Reply:
Hrmmm...looking at the script I think not having the export should be okay?  I was trying to cover the case where the developer wants to override the value.  That won't be available unless it's explicitly exported (or I'm mistaken in understanding the docs).  Either way, I can remove it for now.

Reply:
Also, I don't think it will matter about `realpath` since the mac won't be able to run the container anyway.

BTW, I'm not happy about needing to run a container to lookup the name of the image.  Unf. the script I have that properly interprets the `.cirrus.yml` values is in python.  We can do away with the "run a container" part, but will then need to jump through hoops (or document) to ensure the python 3 interpreter + python3-yaml module are both available :disappointed: 

Okay, I have the export removed in the next version.

Reply:
> I was trying to cover the case where the developer wants to override the value. That won't be available unless it's explicitly exported (or I'm mistaken in understanding the docs).

I couldn’t immediately find anything in `info make` to the contrary, and
```make
with_default = with_default_value

target:
	echo $(no_default)
	echo $(with_default)
```
allows overriding using `make -s no_default=a with_default=b` without an export.

Reply:
> Also, I don't think it will matter about `realpath` since the mac won't be able to run the container anyway.

Yes.

> BTW, I'm not happy about needing to run a container to lookup the name of the image.

\*shrug\* for users that want to run builds/tests in a container, i.e. prefer not to change the current user environment, running the lookup in another container seems likely to be acceptable, whereas imposing requirements on the current user environment might be unwanted; so using a container makes sense to me.

Reply:
Ya, that was my line of thought as well.  Thanks for checking out the Make behavior.  I remember reading the specifics on the official GNU docs page.   But the actual behavior is what matters most.